Update slurm.conf to use cons_tres SelectType

jaygalvin opened the pull request:
The current config yields `slurmctld: fatal: Can't find plugin for select/cons_res`. The solution is to use cons_tres: https://github.com/SchedMD/slurm/commit/0a2905892b3912d5448fc2f3f0e2ca50a8d53231

xuf12 commented:
docker-compose build --no-cache
mysql uses an image, skipping
postgres uses an image, skipping
Building slurmdbd
[+] Building 579.3s (38/46)
 => [internal] load build definition from Dockerfile.compile                                                                                                                                                           0.0s
 => => transferring dockerfile: 45B                                                                                                                                                                                    0.0s
 => [internal] load .dockerignore                                                                                                                                                                                      0.0s
 => => transferring context: 2B                                                                                                                                                                                        0.0s
 => [internal] load metadata for docker.io/library/ubuntu:22.04                                                                                                                                                        0.7s
 => CACHED [ 1/42] FROM docker.io/library/ubuntu:22.04@sha256:4b1d0c4a2d2aaf63b37111f34eb9fa89fa1bf53dd6e4ca954d47caebca4005c2                                                                                         0.0s
 => [internal] load build context                                                                                                                                                                                      0.0s
 => => transferring context: 958B                                                                                                                                                                                      0.0s
 => [ 2/42] COPY rstudio/create.R /tmp/create.R                                                                                                                                                                        0.0s
 => [ 3/42] RUN apt-get update -y &&  apt-get install -y gdebi-core curl wget &&  IFS=" "; for R_VERSION in 3.6.3 4.0.5 4.1.2 ;  do   curl -O https://cdn.rstudio.com/r/ubuntu-2204/pkgs/r-${R_VERSION}_1_amd64.deb  111.8s
 => [ 4/42] COPY rstudio/launcher.conf /etc/rstudio/launcher.conf                                                                                                                                                      0.1s
 => [ 5/42] COPY rstudio/launcher.slurm.conf /etc/rstudio/launcher.slurm.conf                                                                                                                                          0.0s
 => [ 6/42] COPY rstudio/launcher.slurm.profiles.conf /etc/rstudio/launcher.slurm.profiles.conf                                                                                                                        0.0s
 => [ 7/42] COPY rstudio/rserver.conf /etc/rstudio/rserver.conf                                                                                                                                                        0.0s
 => [ 8/42] COPY rstudio/database.conf /etc/rstudio/database.conf                                                                                                                                                      0.0s
 => [ 9/42] RUN chmod 0600 /etc/rstudio/database.conf                                                                                                                                                                  0.4s
 => [10/42] COPY rstudio/load-balancer /etc/rstudio/load-balancer                                                                                                                                                      0.0s
 => [11/42] RUN apt-get update && apt-get install -y uuid &&  apt clean all &&  rm -rf /var/cache/apt                                                                                                                  4.1s
 => [12/42] RUN echo `uuid` > /etc/rstudio/secure-cookie-key &&  chown rstudio-server:rstudio-server   /etc/rstudio/secure-cookie-key &&     chmod 0600 /etc/rstudio/secure-cookie-key                                 0.3s
 => [13/42] RUN openssl genpkey -algorithm RSA   -out /etc/rstudio/launcher.pem   -pkeyopt rsa_keygen_bits:2048 &&  chown rstudio-server:rstudio-server   /etc/rstudio/launcher.pem &&         chmod 0600 /etc/rstudi  0.4s
 => [14/42] RUN openssl rsa -in /etc/rstudio/launcher.pem   -pubout > /etc/rstudio/launcher.pub &&  chown rstudio-server:rstudio-server   /etc/rstudio/launcher.pub                                                    0.4s
 => [15/42] RUN curl -O https://cdn.rstudio.com/python/ubuntu-2204/pkgs/python-3.8.10_1_amd64.deb &&     apt-get update && gdebi -n python-3.8.10_1_amd64.deb && apt clean all &&     rm -rf /var/cache/apt && rm -f  42.7s
 => [16/42] RUN /opt/python/3.8.10/bin/jupyter-nbextension install --sys-prefix --py rsp_jupyter &&     /opt/python/3.8.10/bin/jupyter-nbextension enable --sys-prefix --py rsp_jupyter &&     /opt/python/3.8.10/bin  1.6s
 => [17/42] RUN curl -O https://cdn.rstudio.com/python/ubuntu-2204/pkgs/python-3.9.5_1_amd64.deb &&     apt-get update && gdebi -n python-3.9.5_1_amd64.deb && apt clean all &&     rm -rf /var/cache/apt && rm -f p  36.9s
 => [18/42] RUN /opt/python/3.9.5/bin/jupyter-nbextension install --sys-prefix --py rsp_jupyter &&     /opt/python/3.9.5/bin/jupyter-nbextension enable --sys-prefix --py rsp_jupyter &&     /opt/python/3.9.5/bin/ju  1.5s
 => [19/42] COPY rstudio/jupyter.conf /etc/rstudio/jupyter.conf                                                                                                                                                        0.0s
 => [20/42] RUN rstudio-server install-vs-code /opt/code-server/                                                                                                                                                      10.3s
 => [21/42] COPY rstudio/vscode.conf /etc/rstudio/vscode.conf                                                                                                                                                          0.0s
 => [22/42] COPY rstudio/vscode-user-settings.json  /etc/rstudio/vscode-user-settings.json                                                                                                                             0.0s
 => [23/42] RUN /bin/bash -c "set -x     && groupadd -r --gid=105 munge     && useradd -r -s /bin/bash -g munge --uid=105 munge"                                                                                       0.3s
 => [24/42] RUN set -ex     && apt-get update     && apt-get -y install        wget        bzip2        perl        gcc-9        g++-9        gcc        g++        git        gnupg        make        munge         33.7s
 => [25/42] RUN set -ex     && apt-get update     && apt-get -y install libcgroup-dev     && apt clean all     && rm -rf /var/cache/apt                                                                                4.1s
 => [26/42] RUN mkdir -p /tmp/build && cd /tmp/build     && bash -c "git clone --depth 1 -b slurm-${SLURM_VERSION//./-} https://github.com/SchedMD/slurm.git"     && cd slurm     && bash -c "./configure --enable-  327.3s
 => [27/42] COPY slurm/slurm.conf /etc/slurm/slurm.conf                                                                                                                                                                0.0s
 => [28/42] COPY slurm/slurmdbd.conf /etc/slurm/slurmdbd.conf                                                                                                                                                          0.0s
 => [29/42] RUN echo "CgroupAutomount=yes \nCgroupPlugin=cgroup/v1 \nConstrainCores=yes \nConstrainRAMSpace=yes" > /etc/slurm/cgroup.conf                                                                              0.3s
 => [30/42] RUN chmod 0600 /etc/slurm/slurmdbd.conf                                                                                                                                                                    0.3s
 => [31/42] RUN chown 105:105 /etc/munge/munge.key                                                                                                                                                                     0.4s
 => [32/42] RUN mkdir -p /tmp/singularity                                                                                                                                                                              0.3s
 => [33/42] COPY singularity/spank/* /tmp/singularity/                                                                                                                                                                 0.0s
 => ERROR [34/42] RUN cd /tmp/singularity && make && make install #&& cd && rm -rf /tmp/singularity                                                                                                                    0.6s
------
 > [34/42] RUN cd /tmp/singularity && make && make install #&& cd && rm -rf /tmp/singularity:
#38 0.244 g++ -std=c++17 -I /usr/include/slurm -O2 -Wall -Wextra -fpic -shared -static-libstdc++ -static-libgcc -o singularity-exec.so main.cpp
#38 0.496 main.cpp: In static member function 'static int singularity_exec::start_container(Buttocks, const ArgumentVector&)':
#38 0.496 main.cpp:445:13: warning: this 'if' clause does not guard... [-Wmisleading-indentation]
#38 0.496   445 |             if (env)
#38 0.496       |             ^~
#38 0.496 main.cpp:447:15: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the 'if'
#38 0.496   447 |               s_container_name = s_container_path+"/"+s_container_name;
#38 0.496       |               ^~~~~~~~~~~~~~~~
#38 0.503 main.cpp: At global scope:
#38 0.503 main.cpp:513:44: error: 'SLURM_VERSION_NUMBER' was not declared in this scope; did you mean 'S_SLURM_VERSION_MINOR'?
#38 0.503   513 |   extern const uint32_t plugin_version   = SLURM_VERSION_NUMBER;
#38 0.503       |                                            ^~~~~~~~~~~~~~~~~~~~
#38 0.503       |                                            S_SLURM_VERSION_MINOR
#38 0.594 make: *** [Makefile:11: singularity-exec.so] Error 1
------
executor failed running [/bin/sh -c cd /tmp/singularity && make && make install #&& cd && rm -rf /tmp/singularity]: exit code: 2
ERROR: Service 'slurmdbd' failed to build : Build failed

michaelmayer2 commented:
As discussed, the issue is caused by the inclusion of the SLURM Singularity integration in https://github.com/sol-eng/slurm-docker-cluster/commit/37e7ae73cd132f21703014b49d3d1c63cd882be0. I will accept and merge the PR now and create a separate issue https://github.com/sol-eng/slurm-docker-cluster/issues/3 to make the SLURM integration more flexible/compatible with older SLURM Versions as well.

michaelmayer2 reviewed (approved):
LGTM - thanks for raising this PR

jaygalvin committed:
Update slurm.conf to use cons_tres SelectType

The current config yields `slurmctld: fatal: Can't find plugin for select/cons_res`. The solution is to use cons_tres: https://github.com/SchedMD/slurm/commit/0a2905892b3912d5448fc2f3f0e2ca50a8d53231

michaelmayer2 commented:
Hi. @jaygalvin - thanks for checking out this github repository - I had forked it into the `sol-eng` account and kept developing further over there. The issue you found I actually already fixed in https://github.com/sol-eng/slurm-docker-cluster/commit/97a0402d04d97d65d03617034ea2b8be24994808.

In order to reduce the confusion, I deleted the `sol-eng` fork and transferred the `michaelmayer2/slurm-docker-cluster` repo over to `sol-eng`

michaelmayer2 commented:
Closing - see previous comment.
